app/grandchallenge/cases/views.py

```diff
@@ -142,11 +144,32 @@ def get(self, request, *args, **kwargs):
                 },
             )
             return HttpResponse(html_content)
-        elif current_value and Image.objects.filter(pk=current_value).exists():
-            # this can happen on the display set update view, where one of the options
-            # is the current image, this enables switching back from one of the
-            # above widgets to the chosen image
-            return HttpResponse()
+        elif current_value and (
+            Image.objects.filter(pk=current_value).exists()
+            or UserUpload.objects.filter(pk=current_value).exists()
+        ):
+            # this can happen on the display set update view or redisplay of
+            # form upon validation, where one of the options is the current
+            # image, this enables switching back from one of the above widgets
+            # to the chosen image. In the case of form redisplay on validation
+            # error, current_value will be set in the flexible image widget
+            # select dropdown element with name starting with "WidgetChoice-"
+            # which will not be captured on resubmission, because the
+            # current_value is looked up from the form data element with
+            # interface name (in JobCreateForm and MultipleCIVForm). This make
```

Comment:
What you describe is no longer the case with your changes, right?

Reply:
it is still the case. On redisplay we need a form variable with the prefixed interface slug name to be sent to avoid checking for the "WidgetChoice-" variable.

Reply:
Done